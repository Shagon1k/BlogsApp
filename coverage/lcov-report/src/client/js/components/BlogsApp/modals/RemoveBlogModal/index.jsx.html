<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/client/js/components/BlogsApp/modals/RemoveBlogModal/index.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\..\..\..\index.html">All files</a> / <a href="index.html">src/client/js/components/BlogsApp/modals/RemoveBlogModal</a> index.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">60.98% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>25/41</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">36.36% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>12/33</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/12</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37</td><td class="line-coverage quiet"><span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';<span class="cstat-no" title="statement not covered" ><span class="missing-if-branch" title="if path not taken" >I</span><span class="missing-if-branch" title="else path not taken" >E</span><span class="missing-if-branch" title="else path not taken" >E</span></span>
import Portal from '../../../Portal.jsx'
import './styles.scss';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="branch-1 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="else path not taken" >E</span><span class="branch-1 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="if path not taken" >I</span></span></span></span></span></span></span></span></span></span></span>
&nbsp;
class RemoveBlogModal extends React.Component {
	constructor(props) {<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span>
		super(props);
	}<span class="fstat-no" title="function not covered" ></span>
&nbsp;
	onClose(e) {
		i<span class="cstat-no" title="statement not covered" >f (e) e<span class="cstat-no" title="statement not covered" >.preventDefault();</span></span>
		t<span class="cstat-no" title="statement not covered" >his.props.history.push('/blogs');</span>
	}<span class="fstat-no" title="function not covered" ></span>
&nbsp;
	onConfirm() {
		t<span class="cstat-no" title="statement not covered" >his.props.onConfirm();</span>
		t<span class="cstat-no" title="statement not covered" >his.props.history.push('/blogs');</span>
	}<span class="fstat-no" title="function not covered" ></span>
&nbsp;
	render() {
		r<span class="cstat-no" title="statement not covered" >eturn(</span>
		&lt;Portal&gt;
			&lt;div className="confirmModal"&gt;
				&lt;a href="#" className="closeBtn" onClick={(e) =&gt; this.onClose(e)}&gt; X &lt;/a&gt;<span class="fstat-no" title="function not covered" ></span>
				&lt;span className="message"&gt; Do you really want to delete {this.props.blogTitle} blog?&lt;/span&gt;
				&lt;div className="buttonSet"&gt;
					&lt;button className="btn" onClick={() =&gt; this.onClose()}&gt;No&lt;/button&gt;<span class="fstat-no" title="function not covered" ></span>
					&lt;button className="btn" onClick={() =&gt; this.onConfirm()}&gt;Yes&lt;/button&gt;<span class="fstat-no" title="function not covered" ></span>
				&lt;/div&gt;
			&lt;/div&gt;
		&lt;/Portal&gt;
		);
	}
}
&nbsp;
export default RemoveBlogModal;<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="missing-if-branch" title="else path not taken" >E</span></span></span>
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Mar 03 2018 15:03:49 GMT+0300 (Belarus Standard Time)
</div>
</div>
<script src="..\..\..\..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\..\..\..\sorter.js"></script>
</body>
</html>
