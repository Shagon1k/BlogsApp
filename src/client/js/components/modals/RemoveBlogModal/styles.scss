.modalContainer > div {
	top: 0;
	right: 0;
	left: 0;
	bottom: 0;
	width: 100%;
	height: 100%;
	position: fixed;
	background-color: rgba(0,0,0,0.4);

	.modal {
		margin: 300px auto;
		padding: 15px;
		width: 600px;
		height: auto;
		background: #ccd2dd;
		border: 1px solid black;
	}
	
	.closeBtn {
		position: absolute;
		top: 0;
		right: 0;
		font-size: 18px;
		font-weight: bold;
	}
}

.confirmModal {
	position: relative;

	.message {
		display: inline-block;
		margin: 20px 0;
		width: 100%;
		text-align: center;
		font-size: 18px;
	}

	.buttonSet {
		width: 60%;
		margin: 15px auto;
		display: flex;
		flex-flow: row wrap;
		align-items: center;
		justify-content: space-between;
		
		button {
			display: inline-block;
			width: 125px;
			height: 40px;
			border-radius: 5px;
		}
	}
}