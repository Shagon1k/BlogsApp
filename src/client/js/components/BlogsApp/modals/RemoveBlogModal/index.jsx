import React from 'react';
import Portal from '../../../Portal.jsx'
import './styles.scss';

class RemoveBlogModal extends React.Component {
	constructor(props) {
		super(props);
	}

	onClose(e) {
		if (e) e.preventDefault();
		this.props.history.push('/blogs');
	}

	onConfirm() {
		this.props.onConfirm();
		this.props.history.push('/blogs');
	}

	render() {
		return(
		<Portal>
			<div className="confirmModal">
				<a href="#" className="closeBtn" onClick={(e) => this.onClose(e)}> X </a>
				<span className="message"> Do you really want to delete {this.props.blogTitle} blog?</span>
				<div className="buttonSet">
					<button className="btn" onClick={() => this.onClose()}>No</button>
					<button className="btn" onClick={() => this.onConfirm()}>Yes</button>
				</div>
			</div>
		</Portal>
		);
	}
}

export default RemoveBlogModal;
