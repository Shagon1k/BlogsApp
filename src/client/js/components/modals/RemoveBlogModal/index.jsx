import React from 'react';
import Portal from '../../Portal.jsx'
import './styles.scss';

const RemoveBlogModal = ({onConfirm, onClose, message}) => (
	<Portal>
		<div className="confirmModal">
			<a href="#" className="closeBtn" onClick={() => onClose()}> X </a>
			<span className="message">{message}</span>
			<div className="buttonSet">
				<button className="btn" onClick={() => onClose()}>No</button>
				<button className="btn" onClick={() => onConfirm()}>Yes</button>
			</div>
		</div>
	</Portal>
)

export default RemoveBlogModal;
