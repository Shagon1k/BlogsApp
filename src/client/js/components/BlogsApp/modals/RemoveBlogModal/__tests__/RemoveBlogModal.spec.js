import React from 'react';
import RemoveBlogModal from '../index.jsx';
import { createMemoryHistory } from 'history';

describe('RemoveBlogModal component', () => {
	const onConfirm = jest.fn();
	const dummyBlogTitle = 'Title';
	const history = createMemoryHistory('/dashboard')

	it('renders correctly', () => {
		const wrapper = shallow(<RemoveBlogModal history={history} onConfirm={onConfirm} dummyBlogTitle={dummyBlogTitle}/>);
		expect(wrapper).toMatchSnapshot();

	});
	
	it('triggers onConfirm when button clicked', () => {
		const wrapper = shallow(<RemoveBlogModal history={history} onConfirm={onConfirm} dummyBlogTitle={dummyBlogTitle}/>);
		wrapper.find('.buttonSet button').at(1).simulate('click');
		expect(onConfirm).toBeCalled();
	})
})