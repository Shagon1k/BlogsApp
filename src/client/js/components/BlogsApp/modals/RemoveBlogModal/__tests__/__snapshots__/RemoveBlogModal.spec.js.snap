// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveBlogModal component renders correctly 1`] = `
<Portal>
  <div
    className="confirmModal"
  >
    <a
      className="closeBtn"
      href="#"
      onClick={[Function]}
    >
       X 
    </a>
    <span
      className="message"
    >
       Do you really want to delete 
       blog?
    </span>
    <div
      className="buttonSet"
    >
      <button
        className="btn"
        onClick={[Function]}
      >
        No
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Yes
      </button>
    </div>
  </div>
</Portal>
`;
